utility/src/test/java/org/oppia/android/util/networking/DebugNetworkConnectionUtilTest.kt

```diff
@@ -0,0 +1,278 @@
+package org.oppia.android.util.networking
+
+import android.app.Application
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.logging.EnableConsoleLog
+import org.oppia.android.util.logging.EnableFileLog
+import org.oppia.android.util.logging.GlobalLogLevel
+import org.oppia.android.util.logging.LogLevel
+import org.oppia.android.util.networking.NetworkConnectionUtil.ConnectionStatus
+import org.oppia.android.util.utility.NetworkConnectionTestUtil
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [DebugNetworkConnectionUtil]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class DebugNetworkConnectionUtilTest {
+
+  private val NO_CONNECTION = -1
+
+  @Inject
+  lateinit var debugNetworkConnectionUtil: DebugNetworkConnectionUtil
+
+  @Inject
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerDebugNetworkConnectionUtilTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
+  @Test
+  fun testGetCurrentConnectionStatus_activeWifi_noForcedConnection_returnsWifi() {
+    setNetworkConnectionStatus(
+      status = ConnectivityManager.TYPE_WIFI,
+      networkState = NetworkInfo.State.CONNECTED
+    )
+    assertThat(debugNetworkConnectionUtil.getCurrentConnectionStatus()).isEqualTo(
```

Comment:
Maybe shorten the field to connectionUtil & import the ConnectionStatus constants to simplify things?

Reply:
Done.